### PR TITLE
Redis cache: added parameter to disable SSL certificate verification

### PR DESCRIPTION
**Pull request by dmitsh**
Signed-off-by: Dmitry Shmulevich <dmitry.shmulevich@sysdig.com>

**What this PR does**:
Added configuration parameter to disable SSL certificate verification in Redis client.

**Which issue(s) this PR fixes**:
Related to #3290

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -72,6 +72,7 @@
 * [ENHANCEMENT] Added shuffle sharding support to ruler. Added new metric `cortex_ruler_sync_rules_total`. #3235
 * [ENHANCEMENT] Return an explicit error when the store-gateway is explicitly requested without a blocks storage engine. #3287
 * [ENHANCEMENT] Ruler: only load rules that belong to the ruler. Improves rules synching performances when ruler sharding is enabled. #3269
+* [ENHANCEMENT] Added `redis.tls-skip-verify` flag. Renamed `redis.enable-tls` to `redis.tls-enabled`.
```
Comment (pstibrany) [inline CHANGELOG.md:75]:
```suggestion
* [ENHANCEMENT] Added `redis.tls-skip-verify` flag. Renamed `redis.enable-tls` to `redis.tls-enabled`. #3298
```

**Review by pstibrany — APPROVED** [review]
LGTM, Thanks!

File: CHANGELOG.md [hunk 2]
```diff
@@ -72,6 +72,7 @@
 * [ENHANCEMENT] Added shuffle sharding support to ruler. Added new metric `cortex_ruler_sync_rules_total`. #3235
 * [ENHANCEMENT] Return an explicit error when the store-gateway is explicitly requested without a blocks storage engine. #3287
 * [ENHANCEMENT] Ruler: only load rules that belong to the ruler. Improves rules synching performances when ruler sharding is enabled. #3269
+* [ENHANCEMENT] Added `redis.tls-skip-verify` flag. Renamed `redis.enable-tls` to `redis.tls-enabled`. #3298
```
Comment (pracucci) [inline CHANGELOG.md:75]:
```suggestion
* [CHANGE] Renamed `-redis.enable-tls` CLI flag to `-redis.tls-enabled`. #3298
* [ENHANCEMENT] Added `-redis.tls-skip-verify` flag. #3298
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -72,6 +72,8 @@
 * [ENHANCEMENT] Added shuffle sharding support to ruler. Added new metric `cortex_ruler_sync_rules_total`. #3235
 * [ENHANCEMENT] Return an explicit error when the store-gateway is explicitly requested without a blocks storage engine. #3287
 * [ENHANCEMENT] Ruler: only load rules that belong to the ruler. Improves rules synching performances when ruler sharding is enabled. #3269
+* [CHANGE] Renamed `-redis.enable-tls` CLI flag to `-redis.tls-enabled`. #3298
```
Comment (pracucci) [inline CHANGELOG.md:75]:
```suggestion
* [CHANGE] Renamed `-redis.enable-tls` CLI flag to `-redis.tls-enabled`, and its respective YAML config option from `enable_tls` to `tls_enabled`. #3298
```

File: pkg/chunk/cache/redis_client.go [hunk 4]
```diff
@@ -16,16 +16,17 @@ import (
...
-	MaxConnAge  time.Duration  `yaml:"max_connection_age"`
+	Endpoint      string         `yaml:"endpoint"`
+	MasterName    string         `yaml:"master_name"`
+	Timeout       time.Duration  `yaml:"timeout"`
+	Expiration    time.Duration  `yaml:"expiration"`
+	DB            int            `yaml:"db"`
+	PoolSize      int            `yaml:"pool_size"`
+	Password      flagext.Secret `yaml:"password"`
+	TLSEnabled    bool           `yaml:"tls_enabled"`
+	TLSSkipVerify bool           `yaml:"tls_skip_verify"`
```
Comment (pracucci) [inline pkg/chunk/cache/redis_client.go:27]: For consistency with other similar config, I would name it `tls_insecure_skip_verify`.

File: CHANGELOG.md [hunk 5]
```diff
@@ -72,6 +72,8 @@
 * [ENHANCEMENT] Added shuffle sharding support to ruler. Added new metric `cortex_ruler_sync_rules_total`. #3235
 * [ENHANCEMENT] Return an explicit error when the store-gateway is explicitly requested without a blocks storage engine. #3287
 * [ENHANCEMENT] Ruler: only load rules that belong to the ruler. Improves rules synching performances when ruler sharding is enabled. #3269
+* [CHANGE] Renamed `-redis.enable-tls` CLI flag to `-redis.tls-enabled`, and its respective YAML config option from `enable_tls` to `tls_enabled`. #3298
```
Comment (pracucci) [inline CHANGELOG.md:75]: I added this (because it's a CHANGE). Could you reorder it, moving this line above along with other `[CHANGE]` entries, please?

**Review by pracucci — COMMENTED** [review]
Thanks for working on this! I left a couple of minor comments and then we're good to go.

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Pull request merged**